crates/ansi-to-html/src/color.rs

```diff
@@ -81,6 +84,41 @@ impl Color {
             }
         })
     }
+
+    pub(crate) fn into_opening_fg_span(self, var_prefix: &Option<String>) -> String {
+        self.into_opening_span(var_prefix, true)
+    }
+
+    pub(crate) fn into_opening_bg_span(self, var_prefix: &Option<String>) -> String {
+        self.into_opening_span(var_prefix, false)
+    }
+
+    pub(crate) fn into_opening_span(self, var_prefix: &Option<String>, is_fg: bool) -> String {
+        if let Self::FourBit(four_bit) = self {
+            let mut s = "<span style='".to_owned();
+
+            if is_fg {
+                s.push_str("color");
+            } else {
+                s.push_str("background");
+            }
+
+            s.push_str(":var(--");
+
+            if let Some(prefix) = var_prefix {
+                s.push_str(prefix);
+            }
+
+            four_bit.write_css_var_name(&mut s);
+
+            write!(s, ",{self})'>").expect("String writing is infallible");
+            s
```

Comment:
When `FourBitColor` implements `Display`, all of this can be a simple `format!`.

Reply:
I'm not sure how to make this a single `format!()` without making another struct to hold all the context needed to display and then moving this logic into its display impl. Is that what you're asking for, or am I missing something?